### DISCUSSION TITLE
How do I configure vale with nvim-lint?

### DISCUSSION THREAD

**Discussion by nipuna-perera** [LazyVim/LazyVim#1992] [Q&A] [answered]
Hi I am trying to configure vale but it keeps throwing a parser error unless I set the output to JSON but when I do set it to JSON I don't get any diagnostics in the buffer. Here's my config

```lua
  {
    "mfussenegger/nvim-lint",
    opts = {
      linters_by_ft = {
        go = { "golangcilint" },
        markdown = { "alex", "vale" },
      },
      linters = {
        vale = {
          args = {
            "--config=/myhome/.config/vale/.vale.ini",
            "--output=JSON",
            "--no-exit",
          },
        },
      },
    },
  },

```

Has anyone successfully set up `vale`? My vale.ini file is correct and nvim-lint runs vale fine as I am able to see the lint output when noise shows the parser error.

**Comment by nipuna-perera** [comment, marked as answer]
I figured it out. Since passing `args` overwrites the args config [here](https://github.com/mfussenegger/nvim-lint/blob/master/lua/lint/linters/vale.lua), I basically replicated it in my config like

```lua
local function get_cur_file_extension(bufnr)
  bufnr = bufnr or 0
  return "." .. vim.fn.fnamemodify(vim.api.nvim_buf_get_name(bufnr), ":e")
end

return {
  {
    "mfussenegger/nvim-lint",
    opts = {
      linters_by_ft = {
        go = { "golangcilint" },
        markdown = { "alex", "vale" },
      },
      linters = {
        vale = {
          args = {
            "--config=/path/to/vale/config",
            "--no-exit",
            "--output",
            "JSON",
            "--ext",
            get_cur_file_extension,
          },
        },
      },
    },
  },
}

```

**Reply by dpetka2001** [reply]
You might want to keep an eye on #1887, as if it gets implemented then it would save you the trouble in the future from having to replicate the default config along with your args.

**Comment by nipuna-perera** [comment]
Ohh that would be nice! Thank you!
